static/src/javascripts/projects/common/modules/commercial/creatives/commercial-component.js

```diff
@@ -32,72 +24,54 @@ define([
-            var filteredParams = pick(params || {}, function (v) {
-                    return isArray(v) ? v.length : v;
-                }),
-                query = size(filteredParams) ? '?' + constructQuery(filteredParams) : '';
-            return [config.page.ajaxUrl, '/commercial/', url, '.json', query].join('');
-        },
-        /**
-         * Loads commercial components.
-         * * https://www.google.com/dfp/59666047#delivery/CreateCreativeTemplate/creativeTemplateId=10023207
-         *
-         * @constructor
-         * @extends Component
-         * @param (Object=) adSlot
-         * @param {Object=} params
-         */
-        CommercialComponent = function (adSlot, params) {
-            this.params = params || {};
-            this.type   = this.params.type;
-            // remove type from params
-            delete this.params.type;
-            this.adSlot    = adSlot.length ? adSlot[0] : adSlot;
-            this.components = {
-                bestbuy:        buildComponentUrl('money/bestbuys', this.params),
-                book:           buildComponentUrl('books/book', merge({}, this.params, { t: config.page.isbn || this.params.isbn })),
-                books:          buildComponentUrl('books/books', merge({}, this.params, { t: this.params.isbns ? this.params.isbns.split(',') : [] })),
-                jobs:           buildComponentUrl('jobs', merge({}, this.params, { t: this.params.jobIds ? this.params.jobIds.split(',') : [] }, getKeywords())),
-                masterclasses:  buildComponentUrl('masterclasses', merge({}, this.params, { t: this.params.ids ? this.params.ids.split(',') : [] }, getKeywords())),
-                soulmates:      buildComponentUrl('soulmates/mixed', this.params),
-                soulmatesTest:  buildComponentUrl('soulmates-test/mixed', this.params),
-                soulmatesGroup: buildComponentUrl('soulmates/' + this.params.soulmatesFeedName, this.params),
-                travel:         buildComponentUrl('travel/offers', merge({}, this.params, getKeywords())),
-                multi:          buildComponentUrl('multi', merge({}, this.params, getKeywords())),
-                capiSingle:     buildComponentUrl('capi-single', this.params),
-                capi:           buildComponentUrl('capi', this.params),
-                paidforCard:    buildComponentUrl('paid', this.params)
-            };
+    function defaultUrlBuilder(url) {
+        return function (params) {
+            return buildComponentUrl(url, params);
+        };
+    }
+
+    function bookUrlBuilder(url) {
+        return function (params) {
+            return buildComponentUrl(url, merge(params, { t: config.page.isbn || params.isbn }));
+        };
+    }
+
+    function soulmatesGroupUrlBuilder(url) {
+        return function (params) {
+            return buildComponentUrl(url + params.soulmatesFeedName, params);
         };
+    }
+
+    function complexUrlBuilder(url, withT, withKeywords) {
```

Comment:
`t` is the name of the parameter. I'll change it to `withSpecificID`, that is its real use.